client/question/services/StdOutSeparatorService.js

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+ //
+ // Licensed under the Apache License, Version 2.0 (the "License");
+ // you may not use this file except in compliance with the License.
+ // You may obtain a copy of the License at
+ //
+ //      http://www.apache.org/licenses/LICENSE-2.0
+ //
+ // Unless required by applicable law or agreed to in writing, software
+ // distributed under the License is distributed on an "AS-IS" BASIS,
+ // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ // See the License for the specific language governing permissions and
+ // limitations under the License.
+
+ /**
+  * @fileoverview Service for displaying printed output.
+  */
+
+tie.factory('StdOutSeparatorService', [
+  function() {
+    /**
+     * The separator used in between test cases' stdOut.
+     *
+     * @type {string}
+     **/
+    var separator = "";
+
+    return {
+      /**
+      * Setter method for the separator.
+      *
+      * @param {string} generated Separator generated during preprocessing.
+      */
+      setSeparator: function(generated) {
+        separator = generated;
+      },
+
+      /**
+      * Getter method for the separator.
+      *
+      * @return {string} .
+      */
+      getSeparator: function() {
+        return separator;
+      },
+
+      /**
+      * Returns the corresponding stdOut for the given test case number for
+      * client version.
+      *
+      * @param {Array} stdOut Array of entire stdOut for all test cases.
```

Comment:
From reading this description, I don't understand what this variable represents. I think of stdout as a string; what does it mean for it to be an array? And if it is an array with one element per test case, why isn't the method just `return stdOut[testNum]`?

This might need to be decribed more clearly.

Reply:
Also, this function looks like it's getting into the weeds a bit ... and it's not clear that this service should know about the client/server difference. Ideally we would abstract that away so that services that process the stdout don't have to care about where the stdout comes from.

Would it be helpful to change codeEvalResult.output to an array which contains the stdout per test (or null, if we never get to that test case)? That way, it would already be preprocessed and callers can use it easily after that.

Reply:
Moved this logic to be handled in PythonCodeRunner and associated stdOut with CodeEvalResultObjects rather than having a StdOutSeparatorService.